### DISCUSSION TITLE
Thumbnail from PDF

### DISCUSSION THREAD

**Discussion by kriscr** [pimcore/pimcore#12006] [General]
I cannot generate image preview of the pdf in Pimcore 10.4.0 
`{{ pimcore_pdf("mypdf", {'thumbnail' : 'pdfthumb'}) }}`

In my preview I get a 'Preview in progress or not a valid PDF file'. In Pimcore 6.9 i didn't had problems, but need to do a quick fix before rendering.

```
// Fix preview document
            try {
                if (!$this->pdf("mypdf")->getElement()->getCustomSetting('document_page_count')) {
                    $this->pdf("mypdf")->getElement()->processPageCount();
                }
            } catch (\Exception $e) {
                //do nothing
            }
```

Is there anything to fix it so I could use an image preview of the pdf file?

**Comment by NiklasBr** [comment]
We have a very similar problem (#10841), thumbnails are supposed to be generated by the message queue, but they are never consumed properly.

**Comment by kriscr** [comment]
@NiklasBr Thanks for your input.
When running 
`bin/console messenger:consume pimcore_core pimcore_maintenance pimcore_image_optimize`
and you add a pdf to a document, then the thumbnail of the pdf is been generated.
